### DISCUSSION TITLE
Display Photo Metadata with a right click in Linux Mint

### DISCUSSION THREAD

**Discussion by 100savage** [linuxmint/discussions#149] [Tutorials]
Use the Context menu to display photo metadata with a simple right-click of the mouse.

1. The software needed for this to work is exiv2 and Zenity.
`sudo apt install exiv2 zenity`

2.  Create a file called image-info.nemo_action and place it in .local/share/nemo/actions and copy the below code into the file
Note: The .local folder is hidden and located in your home directory.
```
[Nemo Action]
Name=Display Image Info
Comment=Displays image information for %F
Exec=<image_info.sh "%F">
Icon-Name=emblem-photos
Selection=s
Extensions=png;jpg;jpeg;gif;bmp;svg;avif;jfif;pjpeg;pjp;webp;ico;cur;tif;tiff;JPG
Dependencies=exiv2;zenity;
```
3.  Then create a file called image_info.sh in the .local/share/nemo/actions directory with this code. Make sure you make this file executable.
```
#!/bin/bash

# Check if an image file is provided as an argument
if [ -z "$1" ]; then
    echo "No image file provided"
    exit 1
fi

# Check if the image file exists and is readable
if [ ! -f "$1" ] || [ ! -r "$1" ]; then
    echo "Invalid image file"
    exit 2
fi

# Use exiv2 to extract the image metadata and pipe it to zenity
exiv2 "$1" | zenity --text-info --title="Image Information" --width=700 --height=700
```
4.  Restart nemo in a terminal
```
nemo -q
```
5.   To run this, just open the nemo file manager. Right-click on a photo and select "Display Image Info"

**Comment by mtwebster** [comment]
Fyi, nemo-media-columns extension can also display this stuff in the list view, without needing to right-click.
